### PR TITLE
Added custom hook on host in build is true

### PR DESCRIPTION
**Pull request by isratrade**
@ohadlevy, is the condition host.build? enough to trigger the custom callback, or is there other logic??

### CODE REVIEW

File: app/models/host.rb [hunk 1]
```diff
@@ -24,6 +24,27 @@ class Host < Puppet::Rails::Host
...
 
+  # Define custom hook that can be called in model by magic methods (before, after, around)
+  define_model_callbacks :ready_for_build, :only => :after
+  define_model_callbacks :not_ready_for_build, :only => :after
+  # Custom hooks will be executed after_commit
+  after_commit :run_custom_hook_ready_for_build, :run_custom_hook_not_ready_for_build
+
+  def run_custom_hook_ready_for_build
+    if respond_to?(:old) && old && build? && (build? != old.build?)
+      run_callbacks :ready_for_build do
```
Comment (ohadlevy) [inline app/models/host.rb]: can we have only one after_commit here?

in terms on names, how about

machine finished provisioning

post_build

machine swapped to build mode (excluding new systems)
pre_provision

?

Comment (isratrade) [inline app/models/host.rb]: @ohadlevy, there are two methods called for after_commit, not just one.

**Comment by ohadlevy** [comment]
@isratrade I've squashed/merged and minor changes, please review both code and commit message.. thanks!

**Comment by isratrade** [comment]
@ohadlevy, I don't see any new commit that you made.

**Comment by ohadlevy** [comment]
see
https://github.com/theforeman/foreman/commit/ca166dba271114a9dbf26ead69071457d703fdc7

**Pull request closed without merging**